src/runtime_src/hip/api/hip_memory.cpp

```diff
@@ -215,6 +217,27 @@ namespace xrt::core::hip
     free(host_src);
   }
 
+
+  static void
+  hip_memcpy_host2device_async(hipDeviceptr_t dst, void* src, size_t size, hipStream_t stream)  
+  {
+    if (!src)
+      throw xrt_core::system_error(hipErrorInvalidValue, "Invalid src pointer in hipMemCpyH2DAsync");
+
+    auto hip_mem_dst = memory_database::instance().get_hip_mem_from_addr(dst);
+    if (!hip_mem_dst)
+      throw xrt_core::system_error(hipErrorInvalidValue, "Invalid destination handle in hipMemCpyH2DAsync");
+
+    auto hip_stream = get_stream(stream);
+    if (!hip_stream)
+      throw xrt_core::system_error(hipErrorInvalidValue, "Invalid stream handle in hipMemCpyH2DAsync");
```

Comment:
It would be best to have a helper inline function for these nullptr checks:
```
template <typename T>
inline void
throw_if_null(const T *ptr, const char *msg) {
    if (ptr != nullptr)
        return;
    throw xrt_core::system_error(hipErrorInvalidValue, msg);
} 

```
and call it like this for all the three cases here...

`throw_if_null(hip_mem_dst, "Invalid destination handle in hipMemCpyH2DAsync");`

`throw_if_null(hip_stream, "Invalid stream handle in hipMemCpyH2DAsync");`

...

Reply:
@zhangchiming , we already have throw_if* functions in hip/core/common.h . Please use those. I also didn't use them in my previous hip_memory PR :-( . Can you please replace them aswell.

Reply:
fixed